azure-proto-core/ResourceListOperations.cs

```diff
@@ -60,7 +60,7 @@ public class ResourceListOperations
             //if armfilter == null && listbysub exists for the rp use rp call
             //else use arm list call
             ResourceType type;
-            if (!ArmClient.Registry.TryGetResourceType<T>(out type))
+            if (!ArmClient.Registry.TryGetResourceType<U, T>(out type))
```

Comment:
Per above comments, I think we need an item to remove this dependency on the registry - we could follow several patterns, like always allowing a default constructor of the resource types, or using reflection over a static ResourceType property in each class.